.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,164 @@
+Only two extensions are currently used:
+* [cURL][curl] is only used for better performance over sockets.
+* [JSON][json] for decoding received responses.
+
+[wp-requirements]: https://wordpress.org/about/requirements/
+[semver]: https://semver.org/
+[curl]: https://www.php.net/book.curl
+[json]: https://www.php.net/book.json
+
+## Linting
+
+Code should (obviously) not contain any parse errors.
+This project uses [PHP Parallel Lint][] to monitor this for all supported PHP versions.
+
+The linter can be run locally using `composer lint`.
+
+[PHP Parallel Lint]: https://github.com/php-parallel-lint/PHP-Parallel-Lint
+
+## Coding Style
+
+Please follow the existing coding style and best practices.
+This project uses [PHP_CodeSniffer][] to detect coding standard violations and apply automated fixes (whenever possible).
+
+* All files can be checked for coding standard violations by running `composer checkcs`.
+* Any automatically applicable fixes can be applied by running `composer fixcs`.
+
+[PHP_CodeSniffer]: https://github.com/squizlabs/PHP_CodeSniffer
+
+## Unit Tests
+
+PRs should include unit tests for all changes.
+
+Tests for this library are written using [PHPUnit][] in combination with the [PHPUnit Polyfills][].
+This means that tests can be written for the latest version of PHPUnit (9.x at the time of writing) and still be run on all PHPUnit versions needed to test all supported PHP versions (PHPUnit 5.x - 9.x).
+
+Tests are organized as follows:
+* Each class in the `src` directory has a corresponding subdirectory in the `tests` directory.
+    Example: the tests for the `\WpOrg\Requests\Cookie` class in `src/Cookie.php` can be found in the `tests/Cookie/` subdirectory.
+* Unit tests for a specific method will be in this subdirectory in a file/class matching the method name.
+    Example: unit tests for the `\WpOrg\Requests\Cookie::domain_matches()` method are in the `WpOrg\Requests\Tests\Cookie\DomainMatchesTest` class in the `tests/Cookie/DomainMatchesTest.php` file.
+* Integration tests for a class will be in this subdirectory in a file/class matching the class name.
+    Example: integration tests for the `\WpOrg\Requests\Cookie` class are in the `WpOrg\Requests\Tests\Cookie\CookieTest` class in the `tests/Cookie/CookieTest.php` file.
+
+If at all possible, features **must** be unit tested.
+All tests should have `@covers` tags pointing to the specific code which is being tested.
+Code coverage is monitored for every PR and for the code base as a whole using [CodeCov][].
+
+[PHPUnit]: https://phpunit.readthedocs.io/en/stable/
+[PHPUnit Polyfills]: https://github.com/Yoast/PHPUnit-Polyfills/
+[CodeCov]: https://app.codecov.io/gh/WordPress/Requests/branch/develop
+
+### Prerequisites for running the tests
+
+- [PHP][] >= 5.6
+- [Composer][]
+- [Python 3][]
+- [pip3][]
+- [mitmproxy][] (`pip3 install mitmproxy`)
```

Comment:
Python 3 and `pip3` are technically not dependencies of this project, they are just a means to install `mitmproxy`. If you manage to install it in any other way, I don't think we need Python and `pip` as dependencies, do we?

As an example, you can easily build an executable Docker container called `mitmproxy` that bundles Python 3 so you don't need to install it on your host environment...

Reply:
Which makes me think we should provide something like that to simplify onboarding at some point.

Reply:
This is not a list of dependencies for the project, but a list of the "Prerequisites for running the tests"....

Reply:
Removing pip and leaving the rest for the "_Which makes me think we should provide something like that to simplify onboarding at some point._" action ;-)